test/dist_autograd_test.py

```diff
@@ -258,6 +258,15 @@ def _verify_graph_for_nested_rpc_call(self, ctx):
 
     def _test_graph(self, fn):
         dst_rank = (self.rank + 1) % self.world_size
+
+        if not dist.is_initialized():
```

Comment:
Let's add a comments here to mention that this is needed because this test case calls `dist.barrier()` in the end.

Reply:
Added.

Reply:
Too much duplication. Consider adding a `with run_barrier_on_exit()`, or a test decorator, for example.